src/Plugin/VacancySource/Emply.php

```diff
@@ -168,30 +144,77 @@ public function submitConfigurationForm(array &$form, FormStateInterface $form_s
    *
    * @return array
    *   Array with the source data ready for import.
+   *
+   * @throws \GuzzleHttp\Exception\GuzzleException
+   *   Thrown if the HTTP request to the Emply API fails.
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   *   Thrown if there is an error saving the Vacancy nodes.
+   * @throws \Exception
+   *   Thrown if there is an error creating the Vacancy nodes.
    */
   public function getData() {
     $items = [];
 
+    // Perform the request to fetch vacancies.
     if ($vacancies = $this->doRequest()) {
       foreach ($vacancies as $vacancy) {
         $item = new \stdClass();
-        $item->guid = $vacancy->adid->__toString();
-        $item->languageCode = $this->getLanguageCode($vacancy);
-        $item->createTime = $this->formatEmplyDate($vacancy->published->__toString());
-        $item->advertisementTitle = $this->formatPlainText($vacancy->adheader->__toString());
-        $item->jobTitle = $this->formatPlainText($vacancy->jobtitle->__toString());
-        $item->categoryWorkArea = $this->getEmplyTermFromFacts($vacancy->facts, 'work_area');
-        $item->categoryWorkTime = $this->getEmplyTermFromFacts($vacancy->facts, 'work_time');
-        $item->categoryEmployementType = $this->getEmplyTermFromFacts($vacancy->facts, 'employement_type');
-        $item->categoryDepartment = $this->formatPlainText($vacancy->department->__toString());
-        $item->body = trim($vacancy->ad->__toString());
-        $item->summary = trim($vacancy->adshorttext->__toString());
-        $item->facts = $this->formatEmplyFacts($vacancy->facts, $vacancy->publishingid);
-        $item->advertisementUrl = $vacancy['apply_link']->__toString();
-        $item->applicationUrl = $vacancy['direct_apply_link']->__toString();
-        $item->dueDate = $this->formatEmplyDate($vacancy->deadline->__toString());
-        $item->dueDateTxt = $vacancy->deadlinetext->__toString();
-        $item->workPlace = $this->getEmplyTermFromFacts($vacancy->facts, 'work_place');
+        $item->guid = isset($vacancy['jobId']) ? (string) $vacancy['jobId'] : '';
+        $item->languageCode = $this->determineLanguageCode($vacancy);
+        $item->createTime = isset($vacancy['created']) ? $this->formatEmplyDate($vacancy['created']) : '';
+
+        // Map 'advertisements' -> 'content' to 'advertisementTitle' and 'body'.
+        if (isset($vacancy['advertisements'][0])) {
```

Comment:
It is a bit unstable to rely on the localized title. If the client change the title in Emply will the importer stop working. It would be better to move back to the configured IDs for each of the 4 categories. On Norden are the 3 categories used by Norden the following:

**Department**
jobDetailsId: 65d957f3-2d98-4c3e-a4a6-2ef4f33903e6

**WorkArea**
jobDetailsId: 1c6b340c-259f-48de-ad6d-b2aa847f6031

**EmployementType**
jobDetailsId: 7b734a0d-9bda-4694-b315-8b93f296c5e2

**WorkTime**
Not used on Norden as far as I can see, but lets keep the configuration field for WorkTime and the import of we have a jobDetailsId in the results matching the configured value.

Reply:
@rastepanyan Please, redo this part.

Reply:
Thanks @beltofte ! The configuration field for 'Work Time' and its implementation is restored and again we are using the configured IDs for all categories.